receiver/loadgenreceiver/config.go

```diff
@@ -25,11 +25,16 @@ type (
 	JsonlFile string
 )
 
-// Config defines configuration for HostMetrics receiver.
+// Config defines configuration for loadgen receiver.
 type Config struct {
 	Metrics MetricsConfig `mapstructure:"metrics"`
 	Logs    LogsConfig    `mapstructure:"logs"`
 	Traces  TracesConfig  `mapstructure:"traces"`
+
+	// Concurrency is the amount of concurrency when sending to next consumer.
+	// The concurrent workers share the amount of workload, instead of multiplying the amount of workload,
+	// i.e. loadgenreceiver still sends up to the same MaxReplay limit.
+	Concurrency int `mapstructure:"concurrency"`
```

Comment:
[to reviewer] I'm not sure what's a good name here. I started with `num_workers` but changed to `concurrency`. In apm-perf it is `agent_replicas`, although the meaning is slightly different, because agent_replicas will imply the agents each send a MaxReplay amount of load. This difference is not meaningful in long-runnning scenarios e.g. apmsoak and otelsoak.

We'll also later have to decide on the name of the flag in otelbench. I'm still thinking about `concurrency`, but in apmbench it is called `agents`, but this time in apmbench the number of agents will _share_ b.N (via `b.RunParallel` and `pb.Next()`).